homeassistant/components/plex/server.py

```diff
@@ -3,22 +3,38 @@
 import plexapi.server
 from requests import Session
 
+from homeassistant.components.media_player import DOMAIN as MP_DOMAIN
 from homeassistant.const import CONF_TOKEN, CONF_URL, CONF_VERIFY_SSL
 
-from .const import CONF_SERVER, DEFAULT_VERIFY_SSL
+from .const import (
+    CONF_SERVER,
+    CONF_SERVER_IDENTIFIER,
+    CONF_SHOW_ALL_CONTROLS,
+    CONF_USE_EPISODE_ART,
+    DEFAULT_VERIFY_SSL,
+    DOMAIN as PLEX_DOMAIN,
+    SERVERS,
+)
 from .errors import NoServersFound, ServerNotSpecified
 
 
 class PlexServer:
     """Manages a single Plex server connection."""
 
-    def __init__(self, server_config):
+    def __init__(self, server_config, options=None):
         """Initialize a Plex server instance."""
         self._plex_server = None
         self._url = server_config.get(CONF_URL)
         self._token = server_config.get(CONF_TOKEN)
         self._server_name = server_config.get(CONF_SERVER)
         self._verify_ssl = server_config.get(CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL)
+        self.options = options
+
+    @staticmethod
+    async def async_options_updated(hass, entry):
```

Comment:
Does this callback really belong in the server module. This just updates the options data in hass.data and that is just to provide a connection between the config yaml options and the entry options, right? This isn't necessary to have a working server.

Reply:
Good point, will refactor.